### PR TITLE
SA-1866: Break Down By report missing targeted and opens data

### PR DESCRIPTION
**Pull request by bkemper**
[SA-1866](https://sparkpost.atlassian.net/browse/SA-1866)

### What Changed
 - Changed `<ReportTable />` to use ReportBuilder metrics provided by `useReportBuilderContext` 

### How To Test

- Verify the bug was fixed using the steps to reproduce from the ticket
- Verify the summary report aggregate table does not break because redux state is being shared between both.
- Verify "Save as CSV" button works

### CODE REVIEW

File: src/pages/reportBuilder/components/ReportTable.js [hunk 1]
```diff
@@ -110,16 +108,15 @@ export const ReportTable = props => {
...
       <TableCollection
-        rowKeyName={rowKeyName}
+        rowKeyName={GROUP_CONFIG[groupByKey].keyName}
         columns={getColumnHeaders()}
         getRowData={getRowData()}
         pagination
         defaultPerPage={10}
         rows={tableData}
-        defaultSortColumn={getDefaultSortColumn()}
+        defaultSortColumn={metrics[0].key}
```
Comment (bkemper) [inline src/pages/reportBuilder/components/ReportTable.js:119]: Removed unnecessary function.

File: src/pages/reportBuilder/components/ReportTable.js [hunk 2]
```diff
@@ -131,10 +122,23 @@ export const ReportTable = props => {
...
+            onChange={value => {
+              dispatch(
+                _getTableDataReportBuilder({
+                  groupBy: value,
+                  metrics,
+                  reportOptions: {
+                    ...reportOptions,
+                    filters: Boolean(reportOptions.filters.length)
+                      ? reportOptions.filters
+                      : undefined,
```
Comment (bkemper) [inline src/pages/reportBuilder/components/ReportTable.js:137]: I don't know why this ternary is necessary, so I left it.  More importantly, I pulled this knowledge out of GroupByOption to get it one step closer to a dumb controlled component.

File: src/pages/reportBuilder/components/ReportTable.js [hunk 3]
```diff
@@ -69,35 +68,28 @@ export const ReportTable = props => {
...
+    ));
 
-  const getDefaultSortColumn = () => {
-    return metrics[0].key;
+    return [primaryCol, ...metricCols];
   };
 
   const renderTable = () => {
-    if (groupBy === 'placeholder') {
+    if (!group || metrics.length === 0) {
```
Comment (bkemper) [inline src/pages/reportBuilder/components/ReportTable.js:89]: Added a guard for when no metrics are selected.

File: src/pages/reportBuilder/components/ReportTable.js [hunk 4]
```diff
@@ -69,35 +68,28 @@ export const ReportTable = props => {
...
-  const getRowData = () => {
-    const group = GROUP_CONFIG[groupBy];
-
-    return row => {
-      const filterKey = row[group.keyName];
-      const newFilter =
-        group.label === 'Subaccount'
-          ? getSubaccountFilter(filterKey)
-          : { type: group.label, value: filterKey };
+  const getRowData = row => {
```
Comment (bkemper) [inline src/pages/reportBuilder/components/ReportTable.js:71]: I didn't change the contents of this function, just removed the unnecessary closure.

File: src/pages/reportBuilder/components/ReportTable.js [hunk 5]
```diff
@@ -22,25 +23,23 @@ const tableWrapper = props => {
...
-    tableData = [],
-    tableLoading,
-    subaccounts,
-  } = props;
-
-  //TODO RB CLEANUP: Change value. Default has to be 'aggregate' for now due to sharing reducer with non-hibana summary report
-  const groupBy = props.groupBy === 'aggregate' ? 'placeholder' : props.groupBy;
+    selectors: { selectSummaryMetricsProcessed: metrics },
+    state: reportOptions,
+  } = useReportBuilderContext();
```
Comment (bkemper) [inline src/pages/reportBuilder/components/ReportTable.js:31]: This is the main change.  I switched this component to use ReportBuilder metrics.

**Review by kchu1337 — APPROVED** [review]
The only thing I can see is that if the number of selected metrics is 0, the breakdown table still shows. If you try to select a group by option, there is an api error snackbar.

**Pull request merged**